Xcode project fails to build with example code

Following the instructions [here](https://alt-romes.github.io/posts/2024-04-02-calling-haskell-from-swift.html), I was able to get everything working up without macros. I imported haskell-ffi as a local dependency in xcode but am unable to get the basic example working.

import HaskellFFI

```
struct User: Codable {
    let name: String
    let age: Int
}

@ForeignImportHaskell
func birthday(cconv: HsCallJSON, user: User) -> User { hstub() }
// changed stub() to hstub() because that seems to be what's exported by the package and stub isn't in scope
```


I get a build error: "Command SwiftCompile failed with a nonzero exit code", struggling to get more information out of this because I'm not too familiar with xcode.

Thanks for your write ups btw, very well written.

```
haskell-ffi/Sources/HaskellFFIMacros/ForeignImportHaskellMacro.swift:1:8: error: no such module 'SwiftCompilerPlugin'
import SwiftCompilerPlugin
       ^

** BUILD FAILED **


The following build commands failed:
	SwiftCompile normal arm64 .../haskell-ffi/Sources/HaskellFFIMacros/ForeignImportHaskellMacro.swift (in target 'HaskellFFIMacros' from project 'haskell-ffi')
	SwiftCompile normal arm64 Compiling\ ForeignImportHaskellMacro.swift .../haskell-ffi/Sources/HaskellFFIMacros/ForeignImportHaskellMacro.swift (in target 'HaskellFFIMacros' from project 'haskell-ffi')

```

Found an actual error message, this is weird to me because SwiftCompilerPlugin is definitely included as a build dependency for the package.

--- a/haskell-ffi/Sources/HaskellFFI/HaskellFFI.swift
+++ b/haskell-ffi/Sources/HaskellFFI/HaskellFFI.swift
@@ -33,6 +33,10 @@ public struct HsCallJSON {
     }
 }
 
+public enum HsFFIError: Error {
+    case requiredSizeIs(Int)
+}
+
 public func hstub() -> Never {
     fatalError("Somehow, a stub for a foreign-imported Haskell function was called")
 }

--- a/haskell-ffi/Sources/HaskellFFIMacros/ForeignImportHaskellMacro.swift
+++ b/haskell-ffi/Sources/HaskellFFIMacros/ForeignImportHaskellMacro.swift
@@ -101,9 +101,7 @@ public struct ForeignImportHaskellMacro: PeerMacro {
                     return """
                     // Allocate buffer for result and allocate a pointer to an int with the initial size of the buffer
                     let buf_size = 1024000
-                    enum HsFFIError: Error {
-                        case requiredSizeIs(Int)
-                    }
+                    
                     return try withUnsafeTemporaryAllocation(of: Int.self, capacity: 1) { size_ptr in
                         size_ptr.baseAddress?.pointee = buf_size
                         
@@ -117,7 +115,7 @@ public struct ForeignImportHaskellMacro: PeerMacro {
                                         throw HsFFIError.requiredSizeIs(required_size)
                                     }
                                 }
-                    
+                        
                                 \(decodeAndReturnResult)
                             }
                         } catch HsFFIError.requiredSizeIs(let required_size) {
@@ -139,16 +137,16 @@ public struct ForeignImportHaskellMacro: PeerMacro {
         """
         func \(fname)(\(raw: args)) -> \(retType) {
           \(raw: args.isEmpty ? "" : "let hs_enc = JSONEncoder()")
-          let hs_dec = JSONDecoder()
+          \(raw: retType.description == "UnsafeMutableRawPointer" ? "" : "let hs_dec = JSONDecoder()")
           do {
             \(raw: encodedArgs)
-            return try {
+            return \(raw: pointersToArgs.isEmpty ? "" : "try ") {
               \(raw: pointersToArgs)
-                \(callAndReturn)
+              \(callAndReturn)
               \(raw: closePointersToArgsClosures)
             }()
           } catch {
-              fatalError("Error decoding JSON marshaled from Haskell, probably: \\(error)")
+            fatalError("Error decoding JSON marshaled from Haskell, probably: \\(error)")
           }
         }
         """
@@ -172,6 +170,8 @@ public enum ForeignImportHaskellError: CustomStringConvertible, Error {
         }
     }
 }
+
+
 
 @main
 struct ForeignImportHaskellMacroPlugin: CompilerPlugin {

